### PR TITLE
spring and jetty migration

### PR DESCRIPTION
**Pull request by bbortt**
both dependency updates are "breaking", because the underlying frameworks have migrated to the [jakarta ee 10 release](https://jakarta.ee/release/10/). this is tightly coupled to a follow-up pull-request in citrusframework/citrus-simulator that should update to spring-boot >= 3.2 ([release notes](https://github.com/spring-projects/spring-boot/releases/tag/v3.2.0)).

contains updates of:
* [Spring >= 6.1.0](https://github.com/spring-projects/spring-framework/releases/tag/v6.1.0)
* [Jetty >= 12.0.0](eclipse.dev/jetty/documentation/jetty-12/programming-guide/index.html#pg-migration-11-to-12)

closes https://github.com/citrusframework/citrus/issues/1010.

### CODE REVIEW

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/security/HttpAuthentication.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright 2023 the original author or authors.
+ *  Copyright 2024 the original author or authors.
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/security/HttpAuthentication.java:2]:
```suggestion
 *  Copyright 2023-2024 the original author or authors.
```

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/security/SecurityHandlerFactory.java [hunk 2]
```diff
@@ -200,10 +222,10 @@ private class SimplePropertyUserStore extends PropertyUserStore {
         @Override
         protected void loadUsers() {
             for (org.citrusframework.http.security.User user : users) {
-                Credential credential = Credential.getCredential(user.getPassword());
+                Credential credential = getCredential(user.getPassword());
 
-                String[] roles = IdentityService.NO_ROLES;
-                if (user.getRoles() != null && user.getRoles().length > 0) {
+                String[] roles = new String[0];
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/security/SecurityHandlerFactory.java:227]:
```suggestion
                String[] roles = isNotEmpty(user.getRoles()) ? user.getRoles() : new String[0];
```

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/server/HttpServer.java [hunk 3]
```diff
@@ -202,7 +252,7 @@ protected void startup() {
             handlers.addHandler(contextCollection);
 
             handlers.addHandler(new DefaultHandler());
-            handlers.addHandler(new RequestLogHandler());
+            // TODO: handlers.addHandler(new RequestLogHandler());
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/server/HttpServer.java:255]: this has something to do with logs, I think the difference is... (TBD; add log example)

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/integration/HttpBasicAuthJavaIT.java [hunk 4]
```diff
@@ -110,7 +110,7 @@ public void basicAuthHeader() {
                 .message());
     }
 
-    @CitrusTest
+      @CitrusTest
```
Comment (bbortt) [inline endpoints/citrus-http/src/test/java/org/citrusframework/http/integration/HttpBasicAuthJavaIT.java:113]:
```suggestion
    @CitrusTest
```

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/message/CookieConverterTest.java [hunk 5]
```diff
@@ -1,11 +1,11 @@
 /*
- *    Copyright 2018 the original author or authors
+ *    Copyright 2018-2024 the original author or authors
  *
  *    Licensed under the Apache License, Version 2.0 (the "License");
  *    you may not use this file except in compliance with the License.
  *    You may obtain a copy of the License at
  *
- *        http://www.apache.org/licenses/LICENSE-2.0
+ *        http:// www.apache.org/licenses/LICENSE-2.0
```
Comment (bbortt) [inline endpoints/citrus-http/src/test/java/org/citrusframework/http/message/CookieConverterTest.java:8]: should do a global pattern match for these:

```suggestion
 *        http://www.apache.org/licenses/LICENSE-2.0
```

File: endpoints/citrus-websocket/src/test/java/org/citrusframework/websocket/endpoint/WebSocketClientTest.java [hunk 6]
```diff
@@ -58,41 +63,35 @@ public void testWebSocketClient() throws Exception {
...
+            return null;
         }).when(session).sendMessage(any(org.springframework.web.socket.WebSocketMessage.class));
 
         webSocketEndpoint.createProducer().send(requestMessage, context);
 
         WebSocketMessage responseMessage = (WebSocketMessage) webSocketEndpoint.createConsumer().receive(context, endpointConfiguration.getTimeout());
-        Assert.assertEquals(responseMessage.getPayload(), responseBody);
-        Assert.assertTrue(responseMessage.isLast());
+        assertEquals(responseMessage.getPayload(), responseBody);
+        assertTrue(responseMessage.isLast());
```
Comment (bbortt) [inline endpoints/citrus-websocket/src/test/java/org/citrusframework/websocket/endpoint/WebSocketClientTest.java:95]:
```suggestion
```

File: endpoints/citrus-ws/src/main/java/org/citrusframework/ws/security/SecurityHandlerFactory.java [hunk 7]
```diff
@@ -201,13 +219,14 @@ protected void doStart() throws Exception {
...
+
         @Override
         protected void loadUsers() {
             for (org.citrusframework.ws.security.User user : users) {
-                Credential credential = Credential.getCredential(user.getPassword());
+                Credential credential = getCredential(user.getPassword());
 
-                String[] roles = IdentityService.NO_ROLES;
-                if (user.getRoles() != null && user.getRoles().length > 0) {
+                String[] roles = new String[0];
```
Comment (bbortt) [inline endpoints/citrus-ws/src/main/java/org/citrusframework/ws/security/SecurityHandlerFactory.java:228]: same change as in HTTP. maybe also have a look if something could be centralized.

File: endpoints/citrus-ws/src/main/java/org/citrusframework/ws/server/WebServiceServer.java [hunk 8]
```diff
@@ -165,7 +206,7 @@ protected void startup() {
             handlers.addHandler(contextCollection);
 
             handlers.addHandler(new DefaultHandler());
-            handlers.addHandler(new RequestLogHandler());
+            // TODO: handlers.addHandler(new RequestLogHandler());
```
Comment (bbortt) [inline endpoints/citrus-ws/src/main/java/org/citrusframework/ws/server/WebServiceServer.java:209]: also impact on log output.

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/server/HttpServer.java (hunk 3 above)
Comment (christophd) [reply endpoints/citrus-http/src/main/java/org/citrusframework/http/server/HttpServer.java:255]: yes, I do not recall why this is being added here. So let's remove and see

**Review by christophd — APPROVED** [review]
Many thanks!

When updating Spring core version we may also have a look into other Spring dependencies to make sure we always use the same version (also in transitive dependencies).

You may check with `mvn dependency:tree` and see if no other Spring version is being used as a transitive dependency.

**Comment by bbortt** [comment]
> When updating Spring core version we may also have a look into other Spring dependencies to make sure we always use the same version (also in transitive dependencies).

ok, let's see.. I'll scan other dependencies for updates as well..

**Pull request merged**